### ISSUE TITLE
Clear-Site-Data and sandboxing

### ISSUE THREAD

**Issue by annevk** [w3c/webappsec-clear-site-data#64]
The way the origin is currently derived completely ignores sandboxing. That seems like a bug.

**Comment by jakearchibald** [comment]
Agreed. Putting `Content-Security-Policy: sandbox` on a response should protect your origin from that response. Things like cookie & clear-site-data headers shouldn't operate on the URL origin.

**Comment by annevk** [comment]
I could see the argument that both are set on the server, but if we ever get something like Origin Policy and that could set sandboxing, I'd kinda expect it to be taken into account, so we might as well do it here too.

**Comment by jakearchibald** [comment]
My use-case: I've created an endpoint that proxies content from another site, but adds an ACAO header and sets `Content-Security-Policy: sandbox` (replacing any previous `Content-Security-Policy` header).

I hoped this would make my origin safe from this resource. Maybe things can't be that simple 😄

**Comment by ukusormus** [comment]
Currently, Firefox and Chromium seem to diverge on this issue: Firefox doesn't allow clearing cookies with `Clear-Site-Data` from a sandboxed iframe, while Chromium does.

[PoC](https://response.ee/data/SFRUUC8xLjEgMjAwIE9LDQpDb250ZW50LVR5cGU6IHRleHQvaHRtbDsgY2hhcnNldD11dGYtOA0KDQo8c2NyaXB0Pgpkb2N1bWVudC5jb29raWUgPSAiYT1iIgo8L3NjcmlwdD4KPGlmcmFtZSBzYW5kYm94PSIiIHNyYz0iaHR0cHM6Ly9yZXNwb25zZS5lZS9kYXRhL1NGUlVVQzh4TGpFZ01qQXdJRTlMRFFwRGIyNTBaVzUwTFZSNWNHVTZJSFJsZUhRdmFIUnRiRHNnWTJoaGNuTmxkRDExZEdZdE9BMEtRMnhsWVhJdFUybDBaUzFFWVhSaE9pQWlLaUlOQ2cwS1kyeGxZWEl0YzJsMFpTMWtZWFJoIj48L2lmcmFtZT4KPHNjcmlwdD4Kc2V0VGltZW91dCgoKSA9PiB7CiAgYWxlcnQoIkNvb2tpZXM6ICIgKyBkb2N1bWVudC5jb29raWUpIAp9LCAxMDAwKQo8L3NjcmlwdD4)

(tested only on FF 139.0, Chrome 136.0.7103.113)
